src/Raven.Studio/typescript/components/common/AceEditor.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from "react";
+import { AceEditorMode, LanguageService } from "components/models/aceEditor";
+import { Ace } from "ace-builds";
+import { setCompleters } from "ace-builds/src-noconflict/ext-language_tools";
+import ReactAce, { IAceEditorProps, IAceOptions, ICommand } from "react-ace";
+import "./AceEditor.scss";
+import classNames from "classnames";
+
+export interface AceEditorProps extends IAceEditorProps {
+    mode: AceEditorMode;
+    languageService?: LanguageService;
+    validationErrorMessage?: string;
+    execute?: (...args: any) => any;
+    setIsValid?: (isValid: boolean) => void;
+}
+
+export default function AceEditor(props: AceEditorProps) {
```

Comment:
Did you test if it works with RQL / autocomplete/ JSON etc? We had a lot of customizations in ace editor.

Reply:
I checked for several languages and RQL autocomplete - everything works